src/main/java/ch/wisv/areafiftylan/extras/mailupdates/service/SubscriptionServiceImpl.java

```diff
@@ -0,0 +1,81 @@
+/*
+ *  Copyright (c) 2017  W.I.S.V. 'Christiaan Huygens'
+ *
+ *      This program is free software: you can redistribute it and/or modify
+ *      it under the terms of the GNU General Public License as published by
+ *      the Free Software Foundation, either version 3 of the License, or
+ *      (at your option) any later version.
+ *
+ *      This program is distributed in the hope that it will be useful,
+ *      but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *      MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *      GNU General Public License for more details.
+ *
+ *      You should have received a copy of the GNU General Public License
+ *      along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package ch.wisv.areafiftylan.extras.mailupdates.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import ch.wisv.areafiftylan.exception.SubscriptionNotFoundException;
+import ch.wisv.areafiftylan.extras.mailupdates.model.Subscription;
+import ch.wisv.areafiftylan.extras.mailupdates.model.SubscriptionRepository;
+
+/**
+ * @author Jurriaan Den Toonder Created on 23-10-17
+ */
+@Service
+public class SubscriptionServiceImpl implements SubscriptionService {
+  private final SubscriptionRepository subscriptionRepository;
+
+  @Autowired
+  public SubscriptionServiceImpl(SubscriptionRepository subscriptionRepository) {
+    this.subscriptionRepository = subscriptionRepository;
+  }
+
+  @Override
+  public Subscription addSubscription(String email) {
+    try {
+      return getSubscriptionByEmail(email);
+    } catch (SubscriptionNotFoundException ignored) {}
+
+    Subscription newSubscription = new Subscription(email);
+    return subscriptionRepository.saveAndFlush(newSubscription);
+  }
+
+  @Override
+  public void removeSubscription(Long id) {
+    subscriptionRepository.delete(id);
+  }
+
+  @Override
+  public Collection<Subscription> getSubscriptions() {
+    return subscriptionRepository.findAll();
+  }
+
+  @Override
+  public Subscription getSubscriptionByEmail(String email) {
+    Optional<Subscription> subscription = subscriptionRepository.findByEmail(email);
+    if (!subscription.isPresent()) {
```

Comment:
I think we can use an `orElseThrow` here, which does exactly what you are trying to do here. [Check it out.](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElseThrow-java.util.function.Supplier-)

Reply:
I tried that but it wouldn't work. It was the first thing I wrote, but I got some funky error messages. I am too tired now to fix it though, I'll probably find the issue tomorrow.